internal/workloadinsight/io_renderer.go

```diff
@@ -0,0 +1,274 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workloadinsight
+
+import (
+	"fmt"
+	"math"
+	"strings"
+)
+
+const (
+	blockChar   = "█"
+	emptyChar   = " "
+	labelHeader = "[offset,len)"
+)
+
+// humanReadable formats a byte size into a compact string (KB, MB, GB).
+func humanReadable(size uint64) string {
+	const (
+		KB = 1024
+		MB = KB * 1024
+		GB = MB * 1024
+	)
+	switch {
+	case size >= GB:
+		return fmt.Sprintf("%.1fG", float64(size)/float64(GB))
+	case size >= MB:
+		return fmt.Sprintf("%.1fM", float64(size)/float64(MB))
+	case size >= KB:
+		return fmt.Sprintf("%.1fK", float64(size)/float64(KB))
+	default:
+		return fmt.Sprintf("%dB", size)
+	}
+}
+
+// Range represents a byte range [Start, End).
+type Range struct {
+	Start uint64
+	End   uint64
+}
+
+// Renderer renders I/O byte ranges as ASCII plots to visualize the access patterns.
+type Renderer struct {
+	PlotWidth  int // number of columns used for plotting area
```

Comment:
Should we use un-exported fields? They are probably not gonna be used outside of the package.